react/fields/Dropdown/Dropdown.js

```diff
@@ -0,0 +1,239 @@
+      }
+
+      if (labelFor && id) {
+        return new Error(`\`labelProps.htmlFor\` will be overridden by \`id\` in ${componentName}. Please remove it.`);
+      }
+    },
+    inputProps: (props, propName, componentName) => {
+      const { id, inputProps } = props;
+      const { id: inputId } = inputProps || {};
+
+      if (typeof inputProps !== 'undefined' && typeof inputProps !== 'object') {
+        return new Error(`Invalid prop \`inputProps\` of type \`${typeof inputProps}\` supplied to \`${componentName}\`, expected \`object\`.`);
+      }
+
+      if (inputId && id) {
+        return new Error(`\`inputProps.id\` will be overridden by \`id\` in ${componentName}. Please remove it.`);
+      }
+    },
+    /* eslint-disable consistent-return */
+    help: PropTypes.string,
+    helpProps: PropTypes.object,
+    message: PropTypes.string,
+    messageProps: PropTypes.object,
+    options: PropTypes.arrayOf(
+      PropTypes.shape({
+        value: PropTypes.string.isRequired,
+        label: PropTypes.string.isRequired
+      })
+    ).isRequired,
+    placeholder: PropTypes.string
+  };
+
+  static defaultProps = {
+    id: '',
+    className: '',
+    invalid: false,
+    label: '',
+    help: '',
+    message: '',
+    placeholder: ''
+  };
+
+  constructor() {
+    super();
+
+    this.storeInputReference = this.storeInputReference.bind(this);
+    this.renderLabel = this.renderLabel.bind(this);
+    this.renderSelect = this.renderSelect.bind(this);
+    this.renderHelp = this.renderHelp.bind(this);
+    this.renderMessage = this.renderMessage.bind(this);
+    this.renderIcon = this.renderIcon.bind(this);
+  }
+
+  storeInputReference(input) {
+    if (input !== null) {
+      this.input = input;
+    }
+  }
+
+  renderLabel() {
+    const { label } = this.props;
+
+    if (!label) {
+      return;
+    }
+
+    const { labelProps, id } = this.props;
+    const allLabelProps = {
+      ...combineClassNames(labelProps, styles.label),
+      ...(id ? { htmlFor: id } : {})
+    };
+
+    return (
+      <label {...allLabelProps}>
+        {label}
+      </label>
+    );
+  }
+
+  renderSelect() {
+    const { inputProps = {}, id, options = [], placeholder } = this.props;
+    const inputStyles = classnames({
+      [styles.dropdown]: true,
+      [styles.blankValueSelected]: !inputProps.value
+    });
+    const allInputProps = {
+      ...combineClassNames(inputProps, inputStyles),
+      ...(id ? { id } : {}),
+      ref: this.storeInputReference,
+      value: inputProps.value || ''
+    };
+
+    return (
+      <select {...allInputProps}>
+        <option
+          className={styles.blankOption}
+          value=""
+          disabled={true}>
+          { placeholder }
+        </option>
+        {
+          options.map(({ value, label }) => (
+            <option value={value} key={value}>{ label }</option>
```

Comment:
This is interesting as a value of empty string is used in the placeholder option so that would cause bigger problems then react warnings. Maybe I should add custom prop validation for the options array?